File: 3594-identify-the-largest-outlier-in-an-array/identify-the-largest-outlier-in-an-array.cpp
class Solution {
public:
    int getLargestOutlier(vector<int>& nums) {
        int totalSum = 0;
        unordered_map<int, int> freqMap;

        for (int num : nums) {
            totalSum += num;
            freqMap[num] += 1;
        }

        int maxOutlier = INT_MIN;
        for (int num : nums) {
           int requiredSum = totalSum - num;
            if (requiredSum % 2 != 0) continue;

            int potentialSum = requiredSum / 2;
            freqMap[num] -= 1;
            if(freqMap.find(potentialSum)!=freqMap.end() && freqMap[potentialSum]>0)      {                  maxOutlier = max(maxOutlier, num);
            }
            freqMap[num] += 1;
        }

        return maxOutlier;
    }
};

File: 3594-identify-the-largest-outlier-in-an-array/identify-the-largest-outlier-in-an-array.java
class Solution {
    public int getLargestOutlier(int[] nums) {
        int total=0;
        int outlier= Integer.MIN_VALUE;
        Map<Integer,Integer> freq= new HashMap<>();
        for(int i:nums){
            total+=i;
               freq.put(i, freq.getOrDefault(i, 0) + 1);
        }
        for(int k : nums){
               int sum = total-k;
               if(sum%2!=0) continue;
               int required =sum/2;
               if (freq.containsKey(required)) {
                if (required == k) {
                    if (freq.get(required) > 1)    // need a second copy
                        outlier = Math.max(outlier, k);
                } else {
                    outlier = Math.max(outlier, k);
                }
            }

        }
        return outlier;
        
    }
}